File: VB/WpfReportDesigner_local/LocalReportWizardServiceClient.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports DevExpress.Data.XtraReports.ServiceModel
Imports DevExpress.Data.XtraReports.Wizard
Imports DevExpress.Data.Utils.ServiceModel
Imports DevExpress.Data.XtraReports.DataProviders
Imports DevExpress.Data.XtraReports.ServiceModel.DataContracts

Namespace WpfReportDesigner_local
	Friend Class LocalReportWizardServiceClient
		Inherits LocalReportServiceClient
		Implements IReportWizardServiceClient
		#Region "AddNewReportAsync"
        Public Sub AddNewReportAsync(ByVal model As ReportModel, ByVal asyncState As Object) Implements IReportWizardServiceClient.AddNewReportAsync
            RaiseScalarOperationCompletedEvent(AddNewReportCompletedEvent, ReportService.AddNewReport(model), asyncState)
        End Sub
        Public Event AddNewReportCompleted As EventHandler(Of ScalarOperationCompletedEventArgs(Of String)) Implements IReportWizardServiceClient.AddNewReportCompleted
#End Region
		#Region "GetColumnsAsync"
        Public Sub GetColumnsAsync(ByVal dataSourceName As String, ByVal tableInfo As TableInfo, ByVal asyncState As Object) Implements IReportWizardServiceClient.GetColumnsAsync
            RaiseScalarOperationCompletedEvent(GetColumnsCompletedEvent, ReportService.GetColumns(dataSourceName, tableInfo), asyncState)
        End Sub
        Public Event GetColumnsCompleted As EventHandler(Of ScalarOperationCompletedEventArgs(Of IEnumerable(Of ColumnInfo))) Implements IReportWizardServiceClient.GetColumnsCompleted
		#End Region
		#Region "GetDataSourceDisplayNameAsync"
        Public Sub GetDataSourceDisplayNameAsync(ByVal dataSourceName As String, ByVal dataMember As String, ByVal asyncState As Object) Implements IReportWizardServiceClient.GetDataSourceDisplayNameAsync
            RaiseScalarOperationCompletedEvent(GetDataSourceDisplayNameCompletedEvent, ReportService.GetDataSourceDisplayName(dataSourceName, dataMember), asyncState)
        End Sub
        Public Event GetDataSourceDisplayNameCompleted As EventHandler(Of ScalarOperationCompletedEventArgs(Of String)) Implements IReportWizardServiceClient.GetDataSourceDisplayNameCompleted
		#End Region
		#Region "GetDataSourcesAsync"
        Public Sub GetDataSourcesAsync(ByVal asyncState As Object) Implements IReportWizardServiceClient.GetDataSourcesAsync
            RaiseScalarOperationCompletedEvent(GetDataSourcesCompletedEvent, ReportService.GetDataSources(), asyncState)
        End Sub
        Public Event GetDataSourcesCompleted As EventHandler(Of ScalarOperationCompletedEventArgs(Of IEnumerable(Of DataSourceInfo))) Implements IReportWizardServiceClient.GetDataSourcesCompleted
		#End Region
		#Region "GetItemPropertiesAsync"
        Public Sub GetItemPropertiesAsync(ByVal dataSourceName As String, ByVal dataMember As String, ByVal asyncState As Object) Implements IReportWizardServiceClient.GetItemPropertiesAsync
            RaiseScalarOperationCompletedEvent(GetItemPropertiesCompletedEvent, ReportService.GetItemProperties(dataSourceName, dataMember), asyncState)
        End Sub
        Public Event GetItemPropertiesCompleted As EventHandler(Of ScalarOperationCompletedEventArgs(Of PropertyDescriptorProxy())) Implements IReportWizardServiceClient.GetItemPropertiesCompleted
		#End Region
		#Region "GetListItemPropertiesAsync"
        Public Sub GetListItemPropertiesAsync(ByVal dataSourceName As String, ByVal dataMember As String, ByVal asyncState As Object) Implements IReportWizardServiceClient.GetListItemPropertiesAsync
            RaiseScalarOperationCompletedEvent(GetListItemPropertiesCompletedEvent, ReportService.GetListItemProperties(dataSourceName, dataMember), asyncState)
        End Sub
        Public Event GetListItemPropertiesCompleted As EventHandler(Of ScalarOperationCompletedEventArgs(Of PropertyDescriptorProxy())) Implements IReportWizardServiceClient.GetListItemPropertiesCompleted
		#End Region
		#Region "GetTablesAsync"
        Public Sub GetTablesAsync(ByVal dataSourceName As String, ByVal asyncState As Object) Implements IReportWizardServiceClient.GetTablesAsync
            RaiseScalarOperationCompletedEvent(GetTablesCompletedEvent, ReportService.GetTables(dataSourceName), asyncState)
        End Sub
        Public Event GetTablesCompleted As EventHandler(Of ScalarOperationCompletedEventArgs(Of IEnumerable(Of TableInfo))) Implements IReportWizardServiceClient.GetTablesCompleted
		#End Region
		#Region "GetViewsAsync"
        Public Sub GetViewsAsync(ByVal dataSourceName As String, ByVal asyncState As Object) Implements IReportWizardServiceClient.GetViewsAsync
            RaiseScalarOperationCompletedEvent(GetViewsCompletedEvent, ReportService.GetViews(dataSourceName), asyncState)
        End Sub
        Public Event GetViewsCompleted As EventHandler(Of ScalarOperationCompletedEventArgs(Of IEnumerable(Of TableInfo))) Implements IReportWizardServiceClient.GetViewsCompleted
		#End Region
		#Region "CloseAsync"
        Public Sub CloseAsync() Implements IReportWizardServiceClient.CloseAsync
        End Sub
		#End Region
		#Region "Abort"
        Public Sub Abort() Implements IReportWizardServiceClient.Abort
        End Sub
#End Region
    End Class
End Namespace

File: CS/WpfReportDesigner_local/LocalReportWizardServiceClient.cs
using System;
using System.Collections.Generic;
using DevExpress.Data.XtraReports.ServiceModel;
using DevExpress.Data.XtraReports.Wizard;
using DevExpress.Data.Utils.ServiceModel;
using DevExpress.Data.XtraReports.DataProviders;
using DevExpress.Data.XtraReports.ServiceModel.DataContracts;

namespace WpfReportDesigner_local {
    class LocalReportWizardServiceClient : LocalReportServiceClient, IReportWizardServiceClient {
        #region AddNewReportAsync
        public void AddNewReportAsync(ReportModel model, object asyncState) {
            RaiseScalarOperationCompletedEvent(AddNewReportCompleted, ReportService.AddNewReport(model), asyncState);
        }
        public event EventHandler<ScalarOperationCompletedEventArgs<string>> AddNewReportCompleted;
        #endregion
        #region GetColumnsAsync
        public void GetColumnsAsync(string dataSourceName, TableInfo tableInfo, object asyncState) {
            RaiseScalarOperationCompletedEvent(GetColumnsCompleted, ReportService.GetColumns(dataSourceName, tableInfo), asyncState);
        }
        public event EventHandler<ScalarOperationCompletedEventArgs<IEnumerable<ColumnInfo>>> GetColumnsCompleted;
        #endregion
        #region GetDataSourceDisplayNameAsync
        public void GetDataSourceDisplayNameAsync(string dataSourceName, string dataMember, object asyncState) {
            RaiseScalarOperationCompletedEvent(GetDataSourceDisplayNameCompleted, ReportService.GetDataSourceDisplayName(dataSourceName, dataMember), asyncState);
        }
        public event EventHandler<ScalarOperationCompletedEventArgs<string>> GetDataSourceDisplayNameCompleted;
        #endregion
        #region GetDataSourcesAsync
        public void GetDataSourcesAsync(object asyncState) {
            RaiseScalarOperationCompletedEvent(GetDataSourcesCompleted, ReportService.GetDataSources(), asyncState);
        }
        public event EventHandler<ScalarOperationCompletedEventArgs<IEnumerable<DataSourceInfo>>> GetDataSourcesCompleted;
        #endregion
        #region GetItemPropertiesAsync
        public void GetItemPropertiesAsync(string dataSourceName, string dataMember, object asyncState) {
            RaiseScalarOperationCompletedEvent(GetItemPropertiesCompleted, ReportService.GetItemProperties(dataSourceName, dataMember), asyncState);
        }
        public event EventHandler<ScalarOperationCompletedEventArgs<PropertyDescriptorProxy[]>> GetItemPropertiesCompleted;
        #endregion
        #region GetListItemPropertiesAsync
        public void GetListItemPropertiesAsync(string dataSourceName, string dataMember, object asyncState) {
            RaiseScalarOperationCompletedEvent(GetListItemPropertiesCompleted, ReportService.GetListItemProperties(dataSourceName, dataMember), asyncState);
        }
        public event EventHandler<ScalarOperationCompletedEventArgs<PropertyDescriptorProxy[]>> GetListItemPropertiesCompleted;
        #endregion
        #region GetTablesAsync
        public void GetTablesAsync(string dataSourceName, object asyncState) {
            RaiseScalarOperationCompletedEvent(GetTablesCompleted, ReportService.GetTables(dataSourceName), asyncState);
        }
        public event EventHandler<ScalarOperationCompletedEventArgs<IEnumerable<TableInfo>>> GetTablesCompleted;
        #endregion
        #region GetViewsAsync
        public void GetViewsAsync(string dataSourceName, object asyncState) {
            RaiseScalarOperationCompletedEvent(GetViewsCompleted, ReportService.GetViews(dataSourceName), asyncState);
        }
        public event EventHandler<ScalarOperationCompletedEventArgs<IEnumerable<TableInfo>>> GetViewsCompleted;
        #endregion
        #region CloseAsync
        public void CloseAsync() {            
        }        
        #endregion
        #region Abort
        public void Abort() {
        }
        #endregion
    }
}
